### PR TITLE
Core/GameObject: regenerate GO loot if respawn time passed since last loot generation time and not all items were taken

### LINKED ISSUE #8952: Looting issue with gameobject type 3 for quest items (cause by players)
I would like to report an issue which I have encountered with gameobjects (type 3) which contain items which are required for quests,

an example object is http://www.wowhead.com/object=290 which contains http://www.wowhead.com/item=841 which is required for http://www.wowhead.com/quest=64/the-forgotten-heirloom

But have previously encountered this issue with other gameobjects, the issue is if a player opens the gameobject but does not loot the quest item then another player trys to loot the object it will appear empty, if the player who originally opened it goes back and loots the item then it will work again but this wont always happen and object will end up tagged to that player and noone else will be able to loot quest item until server restart unless the player to which it is tagged loots item.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/104231-looting-issue-with-gameobject-type-3-for-quest-items-cause-by-players?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5

- [x] 3.3.5

**Issues addressed:** Closes #8952

**Tests performed:** Compile and simple ingame test

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObject.h [hunk 1]
```diff
@@ -789,6 +789,8 @@ class TC_GAME_API GameObject : public WorldObject, public GridObject<GameObject>
         void AddLootMode(uint16 lootMode) { m_LootMode |= lootMode; }
         void RemoveLootMode(uint16 lootMode) { m_LootMode &= ~lootMode; }
         void ResetLootMode() { m_LootMode = LOOT_MODE_DEFAULT; }
+        void SetLootGenerationTime() { m_lootGenerationTime = time(NULL); }
```
Comment (DDuarte) [inline src/server/game/Entities/GameObject/GameObject.h]: Should this receive the same treatment as #18910 or it doesn't matter in this case?

Comment (xinef1) [inline src/server/game/Entities/GameObject/GameObject.h]: Should, but in this case its rather irrelevant (duration that is mostly measured in minutes)

**Comment by Tonghost** [comment]
:shipit:

**Comment by Keader** [comment]
nice

**Pull request merged**